### PR TITLE
HATS-340 Modify setup Script to Get Cacerts from Setup Repo

### PR DESCRIPTION
**Pull request by sabrinadowning**
https://jira.boozallencsn.com/browse/HATS-340

### CODE REVIEW

File: setup.ps1 [hunk 1]
```diff
@@ -135,14 +135,40 @@ Write-Host "All software installation tasks completed. Press Enter to continue..
...
-    $cacertsFiles = Get-ChildItem -Path $targetDir -Recurse -Filter "cacerts" | Select-Object -ExpandProperty DirectoryName -Unique
+    Write-Host "Downloading Cacerts..."
+    Invoke-WebRequest -Uri "https://github.com/department-of-veterans-affairs/hats-cucumber-framework-setup/blob/main/cacerts" -OutFile "$targetDir\cacerts"
+    
+    # Check if the cacerts file was downloaded
+    if (-Not (Test-Path -Path "$targetDir\cacerts")) {
+		Write-Error "Failed to download cacerts file."
+	}
+	
+    Write-Host "Adding VA cert to cacerts files..."
```
Comment (AddisonSnyder) [inline setup.ps1:146]: Might want to go with 'replacing cacerts files'

File: setup.ps1 [hunk 2]
```diff
@@ -135,14 +135,40 @@ Write-Host "All software installation tasks completed. Press Enter to continue..
...
+		if ($cacertFile.FullName -eq $targetDir) {
+			Write-Host "Skipping target directory: $($cacertFile.FullName)"
+			continue
+		}
+		
+		$oldCertPath = Join-Path -Path $cacertFile.FullName -ChildPath "cacerts"
+		$newCertPath = Join-Path -Path $targetDir -ChildPath "cacerts"
+		
+		# Check old cacerts exists before attempting to replace it
+		if (Test-Path -Path $oldCertPath {
```
Comment (AddisonSnyder) [inline setup.ps1:164]: Kinda weird to test for the file existence right after finding said file. Is this necessary?

File: setup.ps1 [hunk 3]
```diff
@@ -135,14 +135,40 @@ Write-Host "All software installation tasks completed. Press Enter to continue..
...
-        keytool.exe -import -alias va-certificate-authority -file "$targetDir\VA-Internal-S2-RCA1-v1.pem" -keystore $cacertFile -storepass changeit -noprompt
+		
+		# Skip the target directory to avoid overwrite error
+		if ($cacertFile.FullName -eq $targetDir) {
+			Write-Host "Skipping target directory: $($cacertFile.FullName)"
+			continue
+		}
+		
+		$oldCertPath = Join-Path -Path $cacertFile.FullName -ChildPath "cacerts"
+		$newCertPath = Join-Path -Path $targetDir -ChildPath "cacerts"
```
Comment (AddisonSnyder) [inline setup.ps1:161]: Not sure what the point of having multiple paths of the same file is - does `$cacertFile` not work?

**Review by AddisonSnyder — CHANGES_REQUESTED** [review]

File: setup.ps1 (hunk 2 above)
Comment (sabrinadowning) [reply setup.ps1:164]: I guss it's less of a "verify" and more of a "do this under these conditions"

Comment (sabrinadowning) [reply setup.ps1:164]: replace it only if the path exists (yes even though we just defined it above)

File: setup.ps1 (hunk 3 above)
Comment (sabrinadowning) [reply setup.ps1:161]: $cacertFile should work, it was more for tracking purposes

File: setup.ps1 (hunk 1 above)
Comment (sabrinadowning) [reply setup.ps1]: done

File: setup.ps1 [hunk 4]
```diff
@@ -135,14 +135,39 @@ Write-Host "All software installation tasks completed. Press Enter to continue..
...
+		
+		# Skip the target directory to avoid overwrite error
+		if ($cacertsFile.FullName -eq $targetDir) {
+			Write-Host "Skipping target directory: $($cacertsFile.FullName)"
+			continue
+		}
+		
+		$oldCertPath = Join-Path -Path $cacertsFile.FullName -ChildPath "cacerts"
+		
+		# Check old cacerts exists before attempting to replace it
```
Comment (AddisonSnyder) [inline setup.ps1:162]: Should probably remove this logic, it's redundant. If the cacerts didn't exist, it wouldn't be found in line 149. More lines of code = more complexity = more assumptions/bugs = more maintenance

File: setup.ps1 [hunk 5]
```diff
@@ -135,14 +135,39 @@ Write-Host "All software installation tasks completed. Press Enter to continue..
...
-    # Find all directories containing the specified file
-    $cacertsFiles = Get-ChildItem -Path $targetDir -Recurse -Filter "cacerts" | Select-Object -ExpandProperty DirectoryName -Unique
-    foreach ($cacertFile in $cacertsFiles) {
-        Write-Host "Adding cert to $cacertFile"
-        keytool.exe -import -alias va-certificate-authority -file "$targetDir\VA-Internal-S2-RCA1-v1.pem" -keystore $cacertFile -storepass changeit -noprompt
+    Write-Host "Downloading Cacerts..."
+    Invoke-WebRequest -Uri "https://github.com/department-of-veterans-affairs/hats-cucumber-framework-setup/blob/main/cacerts" -OutFile "$targetDir\cacerts"
+    
+    # Check if the cacerts file was downloaded
+    if (-Not (Test-Path -Path "$targetDir\cacerts")) {
```
Comment (AddisonSnyder) [inline setup.ps1:142]: In what situation would cacerts not exist here _and_ a previous error wasn't thrown in 139?

**Review by AddisonSnyder — CHANGES_REQUESTED** [review]

File: setup.ps1 (hunk 5 above)
Comment (sabrinadowning) [reply setup.ps1:142]: One scenario where cacerts might not exist after download attempt without throwing error at 139 - network issues e.g. temporary loss of internet connectivity that occurs after Invoke-WebRequest command is executed but before the file is fully downloaded, the command may complete without error, but the file may not be saved as expected.

**Review by AddisonSnyder — APPROVED** [review]

File: setup.ps1 (hunk 4 above)
Comment (sabrinadowning) [reply setup.ps1]: removed

File: setup.ps1 (hunk 5 above)
Comment (sabrinadowning) [reply setup.ps1]: removed

**Pull request merged**